src/Features/CSharp/Portable/Testing/CSharpTestMethodFinder.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Features.Testing;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.CSharp.Testing;
+
+[ExportLanguageService(typeof(ITestMethodFinder), LanguageNames.CSharp), Shared]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal class CSharpTestMethodFinder([ImportMany] IEnumerable<ITestFrameworkMetadata> testFrameworks) : AbstractTestMethodFinder<MethodDeclarationSyntax>(testFrameworks)
```

Comment:
Maybe add a comment mentioned this doesn't need to be perfect.

Reply:
I have comments here that address that which I think should be enough - https://github.com/dotnet/roslyn/pull/68889/files#diff-27535fd258bfb04fdaee1c5635b7a15dcb6171dc5ee01fef4e9405802f6a8babR16